staging/src/k8s.io/apiserver/pkg/storage/cacher/watch_cache.go

```diff
@@ -257,6 +265,24 @@ func (w *watchCache) processEvent(event watch.Event, resourceVersion uint64, upd
 	return updateFunc(elem)
 }
 
+// Send a bookmark event with the current resourceVersion
+func (w *watchCache) heartbeat() {
```

Comment:
probably we should just stick with the term bookmark instead of introducing a new term "heartbeat".